nemo/utils/exp_manager.py

```diff
@@ -564,12 +565,26 @@ class NeMoModelCheckpoint(ModelCheckpoint):
     """ Light wrapper around Lightning's ModelCheckpoint to force a saved checkpoint on train_end
     """
 
-    def __init__(self, save_best_model=False, postfix=".nemo", **kwargs):
+    def __init__(self, always_save_nemo=False, save_best_model=False, postfix=".nemo", **kwargs):
         # Parse and store "extended" parameters: save_best model and postfix.
+        self.always_save_nemo = always_save_nemo
         self.save_best_model = save_best_model
         self.postfix = postfix
         # Call the parent class constructor with the remaining kwargs.
         super().__init__(**kwargs)
+    
+    @rank_zero_only
+    def on_save_checkpoint(self, trainer, pl_module):
+        output = super().on_save_checkpoint(trainer, pl_module)
+         
+        if not self.always_save_nemo:
+            return output
+
+        # Load the best model and then re-save it
+        if self.save_best_model:
+            trainer.checkpoint_connector.restore(self.best_model_path, on_gpu=trainer.on_gpu)
```

Comment:
This doesn't seem correct. As we will be restoring during training and if your validation error doesn't decrease in that epoch, it would continuously loop here.

Reply:
I see that you already commented on this issue. Will wait until it to be resolved

Reply:
That problem is fixed below.